express-delete/index.js

```diff
@@ -0,0 +1,45 @@
+const express = require('express');
+const app = express();
+
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+app.get('/api/grades', (req, res) => {
+  var returnArray = [];
+  for (const id in grades) {
+    returnArray.push(grades[id]);
+  }
+  res.json(returnArray);
+});
+
+app.delete('/api/grades/:userId', (req, res) => {
+  for (const id in grades) {
+    if (req.params.userId === id) {
+      delete grades[id];
+    }
+  }
```

Comment:
Since you already have the `req.params.userId`, you can use that directly and skip the loop.

```js
delete grades[req.params.userId];
```